### PR TITLE
fix(gh-workflow): skip comment when push directly

### LINKED ISSUE #5801: [GH workflow] analyze_bundle.txt missed on main branch or the workflow set incorrectly
When I merge PR into main branch, I always received the notification that the workflow failed like [this](https://github.com/reactjs/zh-hans.reactjs.org/actions/runs/4495876442).

I found the PR workflow is different from main branch. The latter doesn't generate comment indeed but it upload it and should be used the other workflow I metion at start.

### [The PR workflow](https://github.com/reactjs/zh-hans.reactjs.org/actions/runs/4495163315/jobs/7908500574):
pr
### [The main workflow](https://github.com/reactjs/zh-hans.reactjs.org/actions/runs/4495862925/jobs/7909947887):
main

### PR DESCRIPTION
**Pull request by awxiaoxian2020**
closes #5801

### CODE REVIEW

File: .github/workflows/analyze.yml [hunk 1]
```diff
@@ -89,3 +110,4 @@ jobs:
 
       # The actual commenting happens in the other action, matching the guidance in
       # https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
+
```
Comment (awxiaoxian2020) [inline .github/workflows/analyze.yml:113]:
```suggestion

```

File: .github/workflows/analyze_comment.yml [hunk 2]
```diff
@@ -49,22 +62,22 @@ jobs:
...
-        if: success() && steps.fc.outputs.comment-id == 0
+        if: success() && steps.fc.outputs.comment-id == 0 && github.event.workflow_run.conclusion == 'success' && steps.get-type.outputs.event-type == 'pull_request'
         with:
           issue-number: ${{ steps.get-comment-body.outputs.pr-number }}
           body: ${{ steps.get-comment-body.outputs.body }}
 
       - name: Update Comment
         uses: peter-evans/create-or-update-comment@v1.4.4
-        if: success() && steps.fc.outputs.comment-id != 0
+        if:  success() && steps.fc.outputs.comment-id != 0 && github.event.workflow_run.conclusion == 'success' && steps.get-type.outputs.event-type == 'pull_request'
```
Comment (awxiaoxian2020) [inline .github/workflows/analyze_comment.yml:80]:
```suggestion
        if: success() && steps.fc.outputs.comment-id != 0 && github.event.workflow_run.conclusion == 'success' && steps.get-type.outputs.event-type == 'pull_request'
```

File: .github/workflows/analyze.yml [hunk 3]
```diff
@@ -88,4 +109,4 @@ jobs:
           path: ./pr_number
 
       # The actual commenting happens in the other action, matching the guidance in
-      # https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
+      # https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
```
Comment (awxiaoxian2020) [inline .github/workflows/analyze.yml:112]:
```suggestion
      # https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

```

**Comment by awxiaoxian2020** [comment]
Hi, @gaearon it seems that `github.event.workflow_run.event` doesn't work. I use a stupid workaround but it works in zh-hans repo.

**Comment by gaearon** [comment]
what was the workaround? sorry i haven’t looked at this yet

**Comment by awxiaoxian2020** [comment]
Oh, @gaearon you update the `comment` workflow and I need look into it to create a new Pull Request for this issue.

**Pull request closed without merging**